package/resource_monitor/resource_monitor/src/query_process_file_descriptor.c

```diff
@@ -0,0 +1,499 @@
+/*
+ * Copyright (C) 2018 Swift Navigation Inc.
+ * Contact: Swift Navigation <dev@swiftnav.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+#define _GNU_SOURCE
+
+#include <stddef.h>
+#include <stdlib.h>
+#include <string.h>
+#include <search.h>
+#include <libsbp/linux.h>
+#include <libpiksi/logging.h>
+#include <libpiksi/util.h>
+#include <libpiksi/runit.h>
+#include <libpiksi/table.h>
+#include "query_sys_state.h"
+#include "resource_monitor.h"
+#include "resource_query.h"
+#include "resmon_common.h"
+
+#define DEBUG_QUERY_FD_TAB
+
+#define ITEM_COUNT 10
+#define MAX_PROCESS_COUNT 2048
+#define MAX_FILE_COUNT 4096
+#define SBP_FRAMING_MAX_PAYLOAD_SIZE 255
+#define LSOF_CMD_BUF_LENGTH 1024 * 16
```

Comment:
Using 1 page of memory (4k) should be sufficient since the `run_command` will re-use the buffer.